browser/skus/skus_sdk_mojom_impl.h

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_BROWSER_SKUS_SKUS_SDK_MOJOM_IMPL_H_
+#define BRAVE_BROWSER_SKUS_SKUS_SDK_MOJOM_IMPL_H_
+
+#include <string>
+
+#include "brave/components/skus/common/skus_sdk.mojom.h"
+
+class SkusSdkService;
+
+namespace content {
+class BrowserContext;
+}  // namespace content
+
+namespace brave_rewards {
+
+class SkusSdkMojomImpl final : public skus::mojom::SkusSdk {
```

Comment:
This name really isn't very helpful. First we can remove `Mojom` from it and also I would drop `Sdk`, but I'm not really clear on what the responsibilities of this class are. Would it not make more sense to have `OrderManager` and `CredentialsManager` or something along those lines?

Reply:
Actually we can remove this class entirely and the service itself can implement the mojo interface just like brave wallet. It's much simpler and cleaner that way.

Reply:
The only problem with putting the Mojom implementation in the service is that the service is supposed to be in the component layer and this is in the browser layer. I can add some proper comments though - I did document the rest of the classes. Let me know!

Reply:
actually see above, this class should not exist

Reply:
In `src/brave/browser/brave_content_browser_client.cc`, I can definitely use a factory to get the service instance. But the only comparable example I'm seeing (where a factory involved) is also having a crappy mojom object (comparable to this one) used just for the purposes of binding via `mojo::MakeSelfOwnedReceiver`

Example usage with wallet:
https://github.com/brave/brave-core/blob/77aacf1a3ab552644c8df7fec6dc69e6163f460a/browser/brave_content_browser_client.cc#L272-L280

Here's a link to the impl for Wallet:
https://github.com/brave/brave-core/blob/77aacf1a3ab552644c8df7fec6dc69e6163f460a/components/brave_wallet/browser/brave_wallet_provider_impl.h

Should we still remove? I don't think I can pass the service to `mojo::MakeSelfOwnedReceiver`

Reply:
Cleaned up with https://github.com/brave/brave-core/pull/10847/commits/780676bbebb2283387763229a4592c4eb4eadb9d - but maybe there's something I'm missing

@kylehickinson shared an example below:
https://github.com/brave/brave-core/blob/dacb8874b08c8cc83ffd1052f15429d494a65ae0/browser/brave_wallet/keyring_controller_factory.cc#L22

Can the service have a different method for returning a mojom handle? I need some help understanding

Reply:
Renaming done in https://github.com/brave/brave-core/pull/10847/commits/0c492c6a95c3b11eb0f4ab4c316510b78a030d19 - will attempt to fix this

Reply:
OK this should be sorted! 😄 Let me know. Besides renaming commit above, also added these to address:
- https://github.com/brave/brave-core/pull/10847/commits/791f0b03062acdb751c210042bb3079b338ea051
- https://github.com/brave/brave-core/pull/10847/commits/2dacad8e9efaca7acf15e66420ba435fdc886644

Reply:
Marking as resolved